ui/src/data-services/models/taxa.ts

```diff
@@ -1,10 +1,24 @@
 export type ServerTaxon = {
-  id: number
+  id: string
   name: string
   rank: string
   parent?: ServerTaxon
+  parents?: ServerTaxon[]
 }
 
+const SORTED_RANKS = [
```

Comment:
@mihow I noticed the ranks were not always sorted (you can for example see this on the occurrence table screenshot you posted in PR description). I added some client side sorting for now, based on this list of ranks. I hope it's correct!

We can skip this if backend plan to sort, but leaving it for now.